app/services/puppet_fact_parser.rb

```diff
@@ -145,6 +145,22 @@ def boot_timestamp
     uptime_seconds.nil? ? nil : (Time.zone.now.to_i - uptime_seconds.to_i)
   end
 
+  def virtual
+    facts['is_virtual']
+  end
+
+  def ram
+    facts['memorysize']
```

Comment:
This is in the form `15.54 GiB`. Do we want to use `memorysize_mb` so it's an integer? May also be possible to use the modern fact and return bytes.
```suggestion
    if facts.key?(:memory)
      facts[:memory][:system][:total_bytes]
    else
      facts['memorysize_mb'] * 1024
    end
```

Reply:
turns out, bytes are too high when it comes to ram for an integer, let's see if storing in gigabytes works, we can always change the storage to other unit, e.g. integer in MB. Though I don't think we need such granularity.

Reply:
My point is that the fact is a *string* which includes the human readable version. I'd go with the `memorysize_mb` fact which is a float.

It's also deprecated and I'd like to be able to support the modern fact as well but I wouldn't block on that.

Reply:
ok, I'm fine with using MB instead, Ruby will cut off the string portion but if you say `memorysize_mb` or `memory` is the new fact, then makes sense to me. In such case, @xprazak2 could you please also incorporate this? Just with the change, we should store in megabytes to avoid integer overflow. We probably don't care about byte-granularity to se don't have to use bigdecimal for this.

Reply:
This line can now be removed
```suggestion
```